pkg/virt-config/deprecation/macvtap.go

```diff
@@ -23,8 +23,14 @@ import (
 	v1 "kubevirt.io/api/core/v1"
 )
 
+const MacvtapGate = "Macvtap" // Deprecated
+
 const MacvtapDiscontinueMessage = "Macvtap network binding is discontinued since v1.3. Please refer to Kubevirt user guide for alternatives."
 
+func init() {
+	RegisterFeatureGate(FeatureGate{Name: MacvtapGate, State: Discontinued, Message: MacvtapDiscontinueMessage, VmiSpecUsed: macvtapApiUsed})
```

Comment:
I don't find this a good pattern. The feature gates should reside in one file with a common and unified descriptions. The code required for registering the FG can be  (actually I would prefer) separated.

Reply:
One of this PR objectives is to centralize the FG core service, which is agnostic of any special named FG, i.e. generic and the feature gates themselves.

Ideally, the named feature gates are decentralized, owned by specific SIGs, subprojects or feature (codebases) and not in a centralized location.
This provides the ability to manage features in a scalable manner, reduce the dependency in the project and avoid collisions on such common files which makes changes very hard.

Said that, each SIG can choose to manage this as they wish, i.e. to centralize it on their side.

I am looking for the choice to be at the one who manages the feature, to have freedom to structure everything in a modular way with enough control to make it easy to maintain.

In general, IMO there needs to be a very good reason to centralize anything. Including constants and registration of FGs.
I recognize there are some scenarios in which it makes sense, but there are rare and usually can be mitigated to resolve the needs in other ways.

Reply:
>Ideally, the named feature gates are decentralized, owned by specific SIGs, subprojects or feature (codebases) and not in a centralized location.
This provides the ability to manage features in a scalable manner, reduce the dependency in the project and avoid collisions on such common files which makes changes very hard.

I think there are 2 different aspects here. The code for the feature and feature gate. I would prefer the later is managed centrally and not by sigs. This will ensure same rules applies to graduation and the user experience is same across sigs/featues

Reply:
I was unable to convince the reviewers and approvers on this subject, so I removed the relevant commit and text.